### PR TITLE
Functional tests - Add test Export Categories to csv

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test Export Categories and check csv file
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `DOWNLOAD_PATH='downloadPath' TEST_PATH="functional/BO/03_catalog/02_categories/07_exportCategories.js" URL_FO=shopUrl/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18149)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const loginCommon = require('@commonTests/loginBO');
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const CategoriesPage = require('@pages/BO/catalog/categories');
+// Test context imports
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_catalog_categories_helperCard';
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js]:
```suggestion
const baseContext = 'functional_BO_catalog_categories_exportCategories';
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.categoriesLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.categoriesPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.categoriesPage.pageTitle);
+  });
+
+  it('should export categories to a csv file', async function () {
+    await this.pageObjects.categoriesPage.exportDataToCsv();
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js:63]:
```suggestion
    await testContext.addContextItem(this, 'testIdentifier', 'exportCategoriesToCsv', baseContext);
    await this.pageObjects.categoriesPage.exportDataToCsv();
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    const fileExist = await files.checkFileExistence('category_', 5000, true, 'csv');
+    await expect(fileExist, 'Export of data has failed').to.be.true;
+  });
+
+  it('should check existence of categories data in csv file', async function () {
+    const numberOfCategories = await this.pageObjects.categoriesPage.getNumberOfElementInGrid();
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js:70]:
```suggestion
    await testContext.addContextItem(this, 'testIdentifier', 'checkCategoriesData', baseContext);
    const numberOfCategories = await this.pageObjects.categoriesPage.getNumberOfElementInGrid();
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js (hunk 1 above)
Comment (SimonGrn) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js]:
```suggestion
const baseContext = 'functional_BO_catalog_categories_exportCategories';
```
Be careful there's a typo in the suggestion by Nesrine

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js]: done

File: tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/02_categories/07_exportCategories.js:63]: done

**Review by Progi1984 — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Pull request merged**